### PR TITLE
Moved MouseButton from msg to script_traits

### LINKED ISSUE #9250: Move MouseButton to script_traits
This is defined in `msg` but used by code that depends on `script_traits`.

Code: `components/msg/constellation_msg.rs`, `components/script_traits/lib.rs`.

### PR DESCRIPTION
**Pull request by DarinM223**
Fixes #9250

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/9263)

### CODE REVIEW

File: components/script_traits/script_msg.rs [hunk 1]
```diff
@@ -2,14 +2,14 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+use MouseButton;
```
Comment (KiChjang) [inline components/script_traits/script_msg.rs]: Have you tested whether this passes `./mach test-tidy`?

**Comment by KiChjang** [comment]
Great work so far! Just a small nit and we can merge this!

**Comment by KiChjang** [comment]
Oh, and this needs a rebase :)

**Comment by DarinM223** [comment]
Thanks! My commit initially didn't work with test-tidy but I changed it so it passes on my computer.

**Comment by KiChjang** [comment]
Please squash the two commits together.

**Comment by KiChjang** [comment]
Ahh, unfortunately this needs a rebase again :(

**Comment by DarinM223** [comment]
Yeah, I think I think I finally got this to work. I think this happened because there was a commit merged after mine that conflicted with mine, so I couldn't resolve it without an extra merge commit which couldn't be squashed :(

**Comment by KiChjang** [comment]
Maybe it's better to use `git rebase upstream/master` next time? That way you don't need to `git merget` and then `git rebase -i HEAD~2`.

Anyway, nice work so far!

@bors-servo r+

**Comment by jdm** [comment]
One missing change in ports/gonk/src/input.rs:
```
src/input.rs:9:5: 9:40 error: unresolved import `msg::constellation_msg::MouseButton`. There is no `MouseButton` in `msg::constellation_msg` [E0432]
src/input.rs:9 use msg::constellation_msg::MouseButton;
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/input.rs:9:5: 9:40 help: run `rustc --explain E0432` to see a detailed explanation
error: aborting due to previous error
Could not compile `b2s`.
```

**Comment by nox** [comment]
```
src/input.rs:9:5: 9:18 error: unresolved import `script_traits::MouseButton`. Maybe a missing `extern crate script_traits`? [E0432]
src/input.rs:9 use script_traits::MouseButton;
                   ^~~~~~~~~~~~~
src/input.rs:9:5: 9:18 help: run `rustc --explain E0432` to see a detailed explanation
error: aborting due to previous error
Could not compile `b2s`.
```

**Comment by DarinM223** [comment]
Not sure how to test gonk locally... do I need to set up the android toolchain?

File: components/script_traits/lib.rs [hunk 2]
```diff
@@ -40,8 +40,8 @@ use gfx_traits::LayerId;
 use ipc_channel::ipc::{IpcReceiver, IpcSender};
 use libc::c_void;
 use msg::constellation_msg::{ConstellationChan, Failure, PipelineId, WindowSizeData};
-use msg::constellation_msg::{Key, KeyModifiers, KeyState, LoadData, MouseButton};
-use msg::constellation_msg::{MozBrowserEvent, PipelineNamespaceId, SubpageId};
+use msg::constellation_msg::{Key, KeyModifiers, KeyState, LoadData, SubpageId};
```
Comment (frewsxcv) [inline components/script_traits/lib.rs]: This `SubpageId` import should come on the next line. Our style checker likes imports to be in alphabetic order

**Comment by nox** [comment]
I think ports/gonk/Cargo.toml should list script_traits as a dependency, but it is brought in as a transitive dependency so I'm not sure it matters.

**Comment by KiChjang** [comment]
@bors-servo r-

Hmm, yes, I think @nox is right. `ports/gonk/Cargo.toml` needs to list `script_traits` as a dependency before it can be used in `main.rs`.

**Comment by jdm** [comment]
The changes to ports/gonk/Cargo.toml now cause ports/gonk/Cargo.lock to be updated while building, so http://build.servo.org/builders/gonk/builds/1294/steps/shell/logs/stdio needs to be applied to the actual PR.

**Comment by KiChjang** [comment]
Thanks for sticking up with this! Hopefully Homu will finally build now...

@bors-servo r+

**Comment by Wafflespeanut** [comment]
browser_host.rs:17:60: 17:71 error: unresolved import `msg::constellation_msg::MouseButton`. There is no `MouseButton` in `msg::constellation_msg` [E0432]
    browser_host.rs:17 use msg::constellation_msg::{self, KeyModifiers, KeyState, MouseButton};
                                                                                  ^~~~~~~~~~~
    browser_host.rs:17:60: 17:71 help: run `rustc --explain E0432` to see a detailed explanation
    error: aborting due to previous error
    Could not compile `embedding`.

**Comment by KiChjang** [comment]
Ok, this means that all the crates under the `ports` directory needs to be checked to see whether they're using `msg::constellation_msg::MouseButton`, and they have to be changed accordingly.

**Comment by KiChjang** [comment]
Sorry, buildbot commands are privileged commands only available to reviewers or people with "try" access. I'm not too familiar with the policy here, @jdm or @Manishearth can fill you in.

@bors-servo r+

**Comment by KiChjang** [comment]
@bors-servo retry
* #9286

**Comment by Wafflespeanut** [comment]
@bors-servo retry

- #9303

**Comment by KiChjang** [comment]
Wow this is unreal, there's lots of merge conflicts.

**Comment by Wafflespeanut** [comment]
... because, a lot of stuff is being moved from `msg` to `script_traits` & `gfx_traits`, and some of those PRs have landed. I bet this conflict occurs only in the `use` statements :)

**Comment by DarinM223** [comment]
Did Homu fail on the css transform test because of the merge conflict?

**Comment by KiChjang** [comment]
No, it's a new intermittent issue.

**Comment by KiChjang** [comment]
@DarinM223 Thanks for going through this lengthy process! I'm sorry it took this long; normally it wouldn't, but there were a lot of people editing the same files that you did at the same time and submitting PRs for that.

**Pull request merged**